api/agent/agent.go

```diff
@@ -764,26 +770,32 @@ func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state
 	}
 	defer container.Close()
 
-	// XXX(reed): we need to timeout the cookie create / prepare since docker client doesn't have timeout anymore,
-	// and handle cookie close having a timed out context it still needs to delete the thing. fun stuff
 	logger := logrus.WithFields(logrus.Fields{"id": container.id, "app_id": call.AppID, "fn_id": call.FnID, "image": call.Image, "memory": call.Memory, "cpus": call.CPUs, "idle_timeout": call.IdleTimeout})
 	ctx = common.WithLogger(ctx, logger)
 
+	ctx, cancel := context.WithTimeout(common.BackgroundContext(ctx), prepTimeout)
+	defer cancel()
+
 	cookie, err := a.driver.CreateCookie(ctx, container)
 	if err != nil {
 		call.slots.queueSlot(&hotSlot{done: make(chan struct{}), fatalErr: err})
 		return
 	}
 
-	defer cookie.Close(ctx)
+	defer func() {
+		ctx, cancel := context.WithTimeout(common.BackgroundContext(ctx), removeTimeout)
+		cookie.Close(ctx)
+		cancel()
+	}()
 
 	err = a.driver.PrepareCookie(ctx, cookie)
```

Comment:
i forgot why create and prepare cookie got split up, maybe it was for cold, seems like we can put humpty dumpty back together again now (not implying that needs to get done here, just observing...)

Reply:
cookie Iface doesn't make sense right now, also wait result. Another PR...  I've modified them here: https://github.com/fnproject/fn/pull/1279, which we can pick and carry into a separate PR later.